### PR TITLE
improving lambda APIs

### PR DESCRIPTION
**Pull request by vigowei**
- `delete_function` will no longer error out (500 internal error) when deleting a function that doesn't exist 
- added [update_event_source_mapping](http://boto3.readthedocs.io/en/latest/reference/services/lambda.html#Lambda.Client.update_event_source_mapping)
- added [delete_event_source_mapping](http://boto3.readthedocs.io/en/latest/reference/services/lambda.html#Lambda.Client.delete_event_source_mapping)
- `list_event_source_mappings` is modified per [AWS doc](https://docs.aws.amazon.com/lambda/latest/dg/API_ListEventSourceMappings.html), it will return all mappings unless `EventSourceArn` and/or `FunctionName` are in the request, in which case only matched items will be returned. This helps the user decide if a specific mapping already exists before calling `update_event_source_mapping`

### CODE REVIEW

File: localstack/mock/apis/lambda_api.py [hunk 1]
```diff
@@ -100,6 +100,25 @@ def add_event_source(function_name, source_arn):
     return mapping
 
 
+def update_event_source(uuid_value, function_name, enabled, batch_size):
+    print('updating event_source for function:  %s', function_name)
```
Comment (whummer) [inline localstack/mock/apis/lambda_api.py]: Please remove the print statement from here.

File: localstack/mock/apis/lambda_api.py [hunk 2]
```diff
@@ -502,6 +542,40 @@ def create_event_source_mapping():
...
+def update_event_source_mapping(mapping_uuid):
+    """ Create new event source mapping
+        ---
+        operationId: 'updateEventSourceMapping'
+        parameters:
+            - name: 'request'
+              in: body
+    """
+    data = json.loads(request.data)
+    # uuid = data.get('UUID')
```
Comment (whummer) [inline localstack/mock/apis/lambda_api.py]: This comment is obsolete, please remove it.

File: localstack/mock/apis/lambda_api.py [hunk 3]
```diff
@@ -502,6 +542,40 @@ def create_event_source_mapping():
     return jsonify(mapping)
 
 
+@app.route('%s/event-source-mappings/<mapping_uuid>' % PATH_ROOT, methods=['PUT'])
+def update_event_source_mapping(mapping_uuid):
+    """ Create new event source mapping
```
Comment (whummer) [inline localstack/mock/apis/lambda_api.py]: Update comment: "Update an existing event source mapping"

File: localstack/mock/apis/lambda_api.py [hunk 4]
```diff
@@ -502,6 +542,40 @@ def create_event_source_mapping():
...
+    function_name = data.get('FunctionName') or ''
+    enabled = data.get('Enabled') or True
+    batch_size = data.get('BatchSize') or 100
+    mapping = update_event_source(mapping_uuid, function_name, enabled, batch_size)
+    return jsonify(mapping)
+
+
+@app.route('%s/event-source-mappings/<mapping_uuid>' % PATH_ROOT, methods=['DELETE'])
+def delete_event_source_mapping(mapping_uuid):
+    """ Create new event source mapping
```
Comment (whummer) [inline localstack/mock/apis/lambda_api.py]: Update comment: "Delete an existing event source mapping"

File: localstack/mock/apis/lambda_api.py [hunk 5]
```diff
@@ -502,6 +542,40 @@ def create_event_source_mapping():
...
+        ---
+        operationId: 'deleteEventSourceMapping'
+    """
+    # uuid = data.get('UUID')
```
Comment (whummer) [inline localstack/mock/apis/lambda_api.py]: This comment is obsolete, please remove it.

**Comment by whummer** [comment]
Many thanks for this pull request. I added a few minor comments.

Can I please ask you to raise the updated pull request against the main repository on Bitbucket: https://bitbucket.org/atlassian/localstack The main reason is that the continuous integration builds are set up to run in Bitbucket Pipelines (https://bitbucket.org/atlassian/localstack/addon/pipelines/home). If you create a fork of the repository on Bitbucket, you can easily enable the Pipelines build for your repo (Settings > Pipelines), which helps us to maintain the code quality, as only pull requests with a green build are accepted and merged.

Also, can you please rebase your branch to the current master branch of this repository? (Looks like there have been some force-pushed changes on master in the meantime) Thanks again.

**Comment by vigowei** [comment]
@whummer do I need use a bitbucket account and fork there to do the new PR? Did you see the conflicts above? I also ran into conflicts when I tried to sync my fork.. looks like some changes were on my fork's but not the official master. Do you want those changes to also be included in the new PR?

**Comment by whummer** [comment]
Moved here: https://bitbucket.org/atlassian/localstack/pull-requests/71/improving-lambda-apis-originally-pr-95-on/diff Thanks

**Pull request closed without merging**